### PR TITLE
refactor: Rename JoinErrors and re-use it

### PR DESCRIPTION
**Pull request by hebasto**
A nice `JoinErrors` utility function was introduced in https://github.com/bitcoin-core/gui/pull/379 by Russell Yanofsky.

This PR renames this function and re-uses it across the code base.

### CODE REVIEW

**Comment by hebasto** [comment]
cc @ryanofsky @jarolrod

**Comment by theStack** [comment]
Concept ACK

**Review by Zero-1729 — COMMENTED** [review]
Concept ACK bb56486a170aacb355f4a973f0cd40ab3918a0cd

**Review by theStack — APPROVED** [review]
Code-review ACK bb56486a170aacb355f4a973f0cd40ab3918a0cd

**Review by Talkless — APPROVED** [review]
utACK bb56486a170aacb355f4a973f0cd40ab3918a0cd

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1328,7 +1329,7 @@ static RPCHelpMan verifychain()
                 "\nVerifies blockchain database.\n",
                 {
                     {"checklevel", RPCArg::Type::NUM, RPCArg::DefaultHint{strprintf("%d, range=0-4", DEFAULT_CHECKLEVEL)},
-                        strprintf("How thorough the block verification is:\n - %s", Join(CHECKLEVEL_DOC, "\n- "))},
```
Comment (ryanofsky) [inline src/rpc/blockchain.cpp:1331]: In commit "refactor: Reuse MakeUnorderedList where possible" (bb56486a170aacb355f4a973f0cd40ab3918a0cd)

This looks good but just want to note small change in behavior here: space is removed before the first "-"

**Review by ryanofsky — APPROVED** [review]
Code review ACK bb56486a170aacb355f4a973f0cd40ab3918a0cd. Nice deduping, thanks for this!

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (hebasto) [reply src/rpc/blockchain.cpp:1331]: Yes. While working on this PR I was wondering what is the purpose of a space before the first "-" :)

It seems ignored:
```
$ src/bitcoin-cli --version
Bitcoin Core RPC client version v22.99.0-bb56486a170a
$ src/bitcoin-cli help verifychain > ~/pr
$ src/bitcoin-cli --version
Bitcoin Core RPC client version v22.99.0-03826aecc56c
$ src/bitcoin-cli help verifychain > ~/master
$ diff ~/master ~/pr  # empty
```

**Pull request merged**